cmd/capibmadm/cmd/vpc/key/delete.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package key
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	logf "sigs.k8s.io/cluster-api/cmd/clusterctl/log"
+
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/vpc"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/options"
+)
+
+type keyDeleteOptions struct {
+	name string
+}
+
+// DeleteCommand vpc key delete command.
+func DeleteCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete",
+		Short: "Delete VPC key",
+		Example: `
+# Create key in VPC
+export IBMCLOUD_API_KEY=<api-key>
+capibmadm vpc key delete key-name --region <region>`,
+	}
+
+	options.AddCommonFlags(cmd)
+	var keyDeleteOption keyDeleteOptions
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		if len(args) < 1 {
+			return fmt.Errorf("key-name name is not provided")
+		}
+
+		keyDeleteOption.name = args[0]
+		if err := deleteKey(keyDeleteOption); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	return cmd
+}
+
+func deleteKey(keyDeleteOption keyDeleteOptions) error {
+	log := logf.Log
+	v1, err := vpc.NewV1Client(options.GlobalOptions.VPCRegion)
```

Comment:
Lets rename `v1` to something more descriptive.

Reply:
`v1` is renamed to `vpcservice`

Reply:
`v1` is rename to `vpcClient`